### PR TITLE
Interpret byte array as primitive using VarHandles

### PR DESCRIPTION
**Pull request by ketanv3**
### Description
This PR replaces manual conversion of `byte[]` to `long|int|short` with VarHandles. This is more readable and performant, since it would ultimately call `Unsafe.unalignedAccess()` to read contiguous memory.

Performance was already fast, so I don't expect much reduction in latency. But no harm in making it **2.7x** faster.
Benchmark code: https://gist.github.com/ketanv3/22fb565dfea619b2307f25b161d2c522
```
Benchmark                           Mode  Cnt           Score          Error  Units
InterpretBitsBenchmark.manual      thrpt    5   543160590.013 ± 17214484.939  ops/s
InterpretBitsBenchmark.varHandle   thrpt    5  1448471902.400 ± 61356063.495  ops/s
```

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [ ] ~Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by andrross** [comment]
Any reason not to add `InterpretBitsBenchmark` to our microbenchmarks in this PR as well?

**Comment by ketanv3** [comment]
> Any reason not to add `InterpretBitsBenchmark` to our microbenchmarks in this PR as well?

@andrross Mainly because the implementation comes straight from JDK and has known performance benefits. Since it's unrelated to OpenSearch anyways and is faster in all situations, adding these benchmarks felt excessive. For analogy, this is like comparing `System.arraycopy(...)` with naive implementation of copying an array element-wise.

**Review by backslasht — APPROVED** [review]
LGTM!

**Comment by andrross** [comment]
> > Any reason not to add `InterpretBitsBenchmark` to our microbenchmarks in this PR as well?
> 
> @andrross Mainly because the implementation comes straight from JDK and has known performance benefits. Since it's unrelated to OpenSearch anyways and is faster in all situations, adding these benchmarks felt excessive. For analogy, this is like comparing `System.arraycopy(...)` with naive implementation of copying an array element-wise.

Thanks @ketanv3, that makes sense. Can you rebase this PR then we can get it merged?

**Review by andrross — APPROVED** [review]

File: libs/core/src/main/java/org/opensearch/core/util/BytesRefUtils.java [hunk 1]
```diff
@@ -103,12 +104,12 @@ public static int sortAndDedup(final BytesRefArray bytes, final int[] indices) {
...
+     * @param bytes in big endian order
+     * @return long value
+     */
     public static long bytesToLong(BytesRef bytes) {
-        int high = (bytes.bytes[bytes.offset + 0] << 24) | ((bytes.bytes[bytes.offset + 1] & 0xff) << 16) | ((bytes.bytes[bytes.offset + 2]
-            & 0xff) << 8) | (bytes.bytes[bytes.offset + 3] & 0xff);
-        int low = (bytes.bytes[bytes.offset + 4] << 24) | ((bytes.bytes[bytes.offset + 5] & 0xff) << 16) | ((bytes.bytes[bytes.offset + 6]
-            & 0xff) << 8) | (bytes.bytes[bytes.offset + 7] & 0xff);
-        return (((long) high) << 32) | (low & 0x0ffffffffL);
+        return (long) BitUtil.VH_BE_LONG.get(bytes.bytes, 0);
```
Comment (reta) [inline libs/core/src/main/java/org/opensearch/core/util/BytesRefUtils.java:113]: I think this is not right, the `bytes.offset` is not taken into account:

```suggestion
        return (long) BitUtil.VH_BE_LONG.get(bytes.bytes, bytes.offset);
```

Comment (ketanv3) [reply libs/core/src/main/java/org/opensearch/core/util/BytesRefUtils.java]: Ah, good catch! Fixed and updated the unit-test to catch this.

**Pull request merged**